server/config.go

```diff
@@ -238,6 +253,10 @@ func (cfg *Config) Check() error {
 		}
 	}
 
+	if cfg.RedisCfg.Endpoint == "" && cfg.RedisCfg.Password != "" {
```

Comment:
will be good to check the boolean flag, if we ever add it

Reply:
@samlaf memstore is specific to overriding the eigenda backend which is why its explicitly set. Redis on the other hand is purely a secondary storage type and only targetable via the fallback and cache slices.

Reply:
Wouldn't we possibly want to use redis as replace the eigenda backend, as you mentioned previously, to get persistence across restarts of the proxy for eg? The distinction still feels confusing to me.

Reply:
nah we never would, the eigenda backend is special and redis is not.

Reply:
comparing memstore and redis here. why does memstore have special status but not redis?

Reply:
@samlaf memstore is a drop in-replacement for the eigenda backend and is a `KeyGeneratedStore` where the backend itself generates lookup keys for some storage entity. This type is only supported for memstore and EigenDA. Redis on the other hand is a `PrecomputedKeyStore` and therefore would be unsuitable for emulating eigenda backend interaction flows.

Reply:
I see..... so what happens if I'm writing a keccak (precomputed key) commitment type? Does eigenda-proxy not handle that since the Put() only takes a value?
Also should we put the different types of stores in separate directories? Like how about we put eigenda and memstore in a `KeyGeneratedStore` dir and redis and s3 in `PrecomputedKeyStore`?

Reply:
thought about it - definitely will need to rework it a bit once we introduce another storage backend type. Will be good to do on the next addition